File: LL/src/Core/Domain/Models/Leaderboards/LeaderboardEntry.cs
namespace Domain.Models.Leaderboards;
public class LeaderboardEntry
{
    public Guid CharacterId { get; set; }
    public string CharacterName { get; set; } = string.Empty;
    public int Rank { get; set; }
    public int Level { get; set; }
    public int Experience { get; set; }
}

File: LL/src/Presentation/ll/src/app/shared/models/Dtos/leaderboard/leaderboardEntry.ts
export interface LeaderboardEntry {
  characterId: string;
  characterName: string;
  level: number;
  experience: number;
  rank: number;
}

export interface LeaderboardColumn<T extends LeaderboardEntry> {
  header: string; // Column header text
  value: (row: T) => string | number; // What to render
  isCharacterName?: boolean;
  cellClass?: string; // Tailwind / CSS classes
  alignRight?: boolean; // Right-align numeric values
}
